buildSrc/src/main/java/com/uber/okbuck/core/model/jvm/JvmTarget.java

```diff
@@ -323,6 +328,58 @@ public static String javaVersion(JavaVersion version) {
     return optionBuilder.build();
   }
 
+  private List<String> readKotlinCompilerArguments() {
+    try {
+      // Note: this is bad juju on Gradle 5.0, which would prefer you get the provider and lazily
+      // eval.
+      // We don't differentiate between test and non-test right now because Android projects don't
+      // support test-only configuration. Non-android projects theoretically can, but let's wait for
+      // someone to ask for that support first as it's not very common.
+      Optional<KotlinCompile> kotlinCompileTask = getProject()
+          .getTasks()
+          .withType(KotlinCompile.class)
+          .stream()
+          .findFirst();
+      if (!kotlinCompileTask.isPresent()) {
+        return Collections.emptyList();
+      }
+      ImmutableList.Builder<String> optionBuilder = ImmutableList.builder();
+      KotlinJvmOptions options = kotlinCompileTask.get().getKotlinOptions();
+      optionBuilder.addAll(options.getFreeCompilerArgs());
+
+      // Args from CommonToolArguments.kt and KotlinCommonToolOptions.kt
+      if (options.getAllWarningsAsErrors()) {
+        optionBuilder.add("-Werror");
+      }
+      if (options.getSuppressWarnings()) {
+        optionBuilder.add("-nowarn");
+      }
+      if (options.getVerbose()) {
+        optionBuilder.add("-verbose");
+      }
+
+      // Args from K2JVMCompilerArguments.kt and KotlinJvmOptions.kt
+      optionBuilder.add("-jvm-target", options.getJvmTarget());
+      optionBuilder.add("-include-runtime", Boolean.toString(options.getIncludeRuntime()));
+      String jdkHome = options.getJdkHome();
+      if (jdkHome != null) {
+        optionBuilder.add("-jdk-home", jdkHome);
+      }
+      optionBuilder.add("-no-jdk", Boolean.toString(options.getNoJdk()));
+      optionBuilder.add("-no-stdlib", Boolean.toString(options.getNoStdlib()));
+      optionBuilder.add("-no-reflect", Boolean.toString(options.getNoReflect()));
+      optionBuilder.add("-java-parameters", Boolean.toString(options.getJavaParameters()));
+
+      // In the future, could add any other compileKotlin configurations here
+      return optionBuilder.build().stream()
+          .distinct() // Defensively de-dupe because you could add any options args to free args too
```

Comment:
this should not be here right?

it would change the below
```
        '-no-jdk',
        'true',
        '-no-stdlib',
        'true',
```
to 
```
        '-no-jdk',
        '-no-stdlib',
        'true',
```

Reply:
Ooooo good point. We can just remove the defensive de-dupe for now then

Reply:
we probably need a Pair<String, Optional<String>> which can then be deduped.

Reply:
2bc1ccd

Reply:
Can look at the pair option separately, want to get your thoughts on the build failure first. My buck-fu is not strong enough to know how to get around its esoteric failures